### ISSUE TITLE
terraform plan reports an error with an existing kms policy

### ISSUE THREAD

**Issue by aokojshdgop8791** [hashicorp/terraform-provider-aws#26747]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

Terraform version: v0.13.6
aws provider version: ~> 3.73.0

### Affected Resource(s)

* aws_kms_key

### Terraform Configuration Files

```hcl
resource "aws_kms_key" "test_key" {
   description = "test key"

    policy = <<EOF
{
    "Sid": "Allow Amazon S3 use of the KMS key",
    "Effect": "Allow",
    "Principal": {
        "Service": "s3.amazonaws.com"
    },
    "Action": [
        "kms:GenerateDataKey"
    ],
    "Resource": "*",
    "Condition":{
      "StringEquals":{
         "aws:SourceAccount":"${data.aws_caller_identity.current.account_id}"
     },
      "ArnLike":{
        "aws:SourceARN": [
          "arn:aws:s3:::bucket01",
          "arn:aws:s3:::bucket02"
        ]
     }
   }
}
EOF
}
```

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Debug Output

```shell
Error: while setting policy ({"Statement":[{"Action":"kms:*","Effect":"Allow","Principal":{"AWS":"arn:aws:iam::1234567890:root"},"Resource":"*","Sid":"Allow management of the key"}],"Version":"2012-10-17"}), encountered: Error parsing policy: Unknown error parsing statement
```

The error is due to the incorrect policy defined in the aws_kms_key.  However, even after the policy document is fixed from the console, terraform plan returned the corrected policy instead of the policy in the state file which is malformed. 

### Expected Behavior

terraform plan should be able to parse the above kms policy

### Actual Behavior

terraform plan could not parse the working kms policy

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by justinretzolk** [comment]
Hey @aokojshdgop8791 👋 Thank you for taking the time to raise this! I see you're using a bit of an older version of the AWS Provider. Are you able to test with a newer version of the provider, given that there's been a few bug fixes related to the `aws_kms_key` since `3.73.0`?

If you're not able to, or if the issue persists with the later version, can you supply (redacted as necessary) full debug logs as well?

**Comment by aokojshdgop8791** [comment]
Hi Justin,
   Will not be able to update to the latest version for a short while.   But I got it resolved by removing the resource from the state.  Let me close this for now. Thank you. 

- Wilfred
